projects/cloud/app/services/change_user_role_service.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class ChangeUserRoleService < ApplicationService
+  attr_reader :user_id, :organization_id, :current_role, :new_role, :current_user
+
+  module Error
+    Unauthorized = Class.new(StandardError)
+  end
+
+  def initialize(user_id:, organization_id:, current_role:, new_role:, current_user:)
+    super()
+    @user_id = user_id
+    @organization_id = organization_id
+    @current_role = current_role
+    @new_role = new_role
+    @current_user = current_user
+  end
+
+  def call
+    return user if current_role == new_role
+    ActiveRecord::Base.transaction do
+      organization = Organization.find(organization_id)
+      raise Error::Unauthorized unless OrganizationPolicy.new(current_user, organization).update?
+      user = User.find(user_id)
+      user.remove_role(current_role, organization)
```

Comment:
I think we should design this service to take whatever role the user has to the new role. We can fetch the roles from the user that we fetched from the database.

Reply:
Could you rephrase what you mean?

Currently, roles for a given `organization` are exclusionary - a `User` model can only have a role of an `admin`. `User` cannot be `admin` _and_ `user` at the same time. The other approach could be `User` would have `user` _and_ `admin` role. I am not 100 % sure what's the preferred practice, I've seen both approaches when searching for this.